src/Aspire.Hosting/Nats/NatsBuilderExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding NATS resources to the application model.
+/// </summary>
+public static class NatsBuilderExtensions
+{
+    /// <summary>
+    /// Adds a NATS server resource to the application model. A container is used for local development.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port for NATS server.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<NatsServerResource> AddNats(this IDistributedApplicationBuilder builder, string name, int? port = null)
+    {
+        var nats = new NatsServerResource(name);
+        return builder.AddResource(nats)
+                      .WithAnnotation(new EndpointAnnotation(ProtocolType.Tcp, port: port, containerPort: 4222))
+                      .WithAnnotation(new ContainerImageAnnotation { Image = "nats", Tag = "2" })
```

Comment:
`WithContainerImage`

Reply:
builder.AddResource(nats)
           .WithEndpoint(containerPort: 4222, hostPort: port)
           .WithImage("nats")
           .WithImageTag("2")

Reply:
done. also notice change in [ContainerResourceBuilderExtensions.cs](https://github.com/dotnet/aspire/pull/1175/files#diff-3ab0bf8f6bfe27862cac4d1067b22ba2d2b620c6bfb15d679aeade0130101670)

Reply:
Yea, it seems like we have some bugs. You can revert to using the annotation for now and that change. See https://github.com/dotnet/aspire/pull/2588